insights/parsers/teamdctl_state_dump.py

```diff
@@ -0,0 +1,69 @@
+"""
+teamdctl {team interface} state dump - Command
+==============================================
+
+This module provides processing for the output of the command
+``teamdctl {team interface} state dump`` which is JSON pattern.
+
+Examples:
+
+    >>> teamdctl_state_dump_content = '''
+    ... {
+    ...     "runner": {
+    ...         "active_port": "eno1"
+    ...     },
+    ...     "setup": {
+    ...         "daemonized": false,
+    ...         "dbus_enabled": true,
+    ...         "debug_level": 0,
+    ...         "kernel_team_mode_name": "activebackup",
+    ...         "pid": 4464,
+    ...         "pid_file": "/var/run/teamd/team0.pid",
+    ...         "runner_name": "activebackup",
+    ...         "zmq_enabled": false
+    ...     },
+    ...     "team_device": {
+    ...         "ifinfo": {
+    ...             "dev_addr": "2c:59:e5:47:a9:04",
+    ...             "dev_addr_len": 6,
+    ...             "ifindex": 5,
+    ...             "ifname": "team0"
+    ...         }
+    ...     }
+    ... }
+    ... '''.strip()
+
+    >>> from insights.parsers.teamdctl_state_dump import TeamdctlStateDump
+    >>> from insights.tests import context_wrap
+    >>> shared = {TeamdctlStateDump: TeamdctlStateDump(context_wrap(teamdctl_state_dump_content))}
+    >>> result = TeamdctlStateDump(context_wrap(teamdctl_state_dump_content)).data
+    >>> result['runner']['active_port']
+    eno1
+"""
+
+import json
+from .. import Parser, parser, LegacyItemAccess
+
+
+class TeamdJsonParsing(LegacyItemAccess, Parser):
+    """Base class implementing shared code."""
+
+    def parse_content(self, content):
+        """
+        Parse the output of the ``teamdctl {team interface} state dump``.
+        """
+        self.data = json.loads(''.join(content))
+
+
+@parser("teamdctl_state_dump")
+class TeamdctlStateDump(TeamdJsonParsing):
```

Comment:
Is there a need for this extra class?  If you are not planning on using the base `TeamdJsonParsing` class for other parsers then it makes more sense to just have one class here.